pkg/utils/cryptoConfig.go

```diff
@@ -78,3 +69,21 @@ func CreateCipherCtx(cmnCryptoCfg *CommonCryptoConfig) (*evecommon.CipherContext
 
 	return cipherCtx, nil
 }
+
+// AddCipherCtxToDev add cipher context to device, unless it already exists
+func AddCipherCtxToDev(dev *device.Ctx, cipherCtx *evecommon.CipherContext) {
+	// check if we already have cipherCtx with the same certificates
+	appendCipherCtx := true
+	for _, c := range dev.GetCipherContexts() {
+		sameCipherCtx := bytes.Equal(c.DeviceCertHash, cipherCtx.DeviceCertHash) &&
+			bytes.Equal(c.ControllerCertHash, cipherCtx.ControllerCertHash)
+		if sameCipherCtx {
+			cipherCtx = c
```

Comment:
and then appendCipherCtx won't be needed at all, so it'll look something like,

```go
func AddCipherCtxToDev(dev *device.Ctx, cipherCtx *evecommon.CipherContext) {
	// check if we already have cipherCtx with the same certificates
	for _, c := range dev.GetCipherContexts() {
                sameCipherCtx := CompareSlices(c.DeviceCertHash, cipherCtx.DeviceCertHash) && CompareSlices(c.ControllerCertHash, cipherCtx.ControllerCertHash)
		if sameCipherCtx {
			cipherCtx = c
                        return
		}
	}
       
       dev.SetCipherContexts(append(dev.GetCipherContexts(), cipherCtx))
}
```

Reply:
Absolutely! I think it would be even cleaner to return the cipher context that was added or the matching existing one, since we don't compare all the fields and there might potentially be discrepancies